zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -507,10 +508,11 @@ protected boolean jobAbort() {
     Job job = scheduler.removeFromWaitingQueue(getId());
     if (job != null) {
       job.setStatus(Status.ABORT);
+      aborted = false;
```

Comment:
this seems a bit counter intuitive - if the job is removed from waiting queue and its status is set to abort, isn't it aborted? perhaps add a comment here to explain why it is needed?

Reply:
I agree it seems counter intuitive. Basically, the issue was that in `JobRunner` we check if a job is aborted, then just toggle the flag, set status and return. Below is the code:
```
public void run() {
      if (job.isAborted()) {
        job.setStatus(Status.ABORT);
        job.aborted = false;

        synchronized (queue) {
          running.remove(job);
          queue.notify();
        }
        jobSubmittedRemotely = true;
        return;
      }
```

The best way would have been to be able to remove the above check but I am not sure what side-effects will it have (it could lead to race conditions). The next best way I could think of to avoid this problem was to disable `aborted` so that next run of the paragraph passes.

Reply:
@felixcheung Does the comment above help?